### PR TITLE
allow specifying the next player number in end_turn #1392

### PR DESCRIPTION
**Pull request by gfgtdf**
as title says.

Old post:
> doesn't work yet since the wesnothd code still needs to be updated. also i'm still unsure about the best api, i coul eigher make it a parameter to wesnoth.end_turn or add  seperate function set_next_playerr_number

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
The parameter to `end_turn` seems to make more sense to me.

**Comment by GregoryLundberg** [comment]
So, is it possible for the end-turn for side 1 to specify side 3, and side 2 never gets a turn?

**Comment by gfgtdf** [comment]
> So, is it possible for the end-turn for side 1 to specify side 3, and side 2 never gets a turn?

yes that's the point of this pr

File: src/play_controller.cpp [hunk 1]
```diff
@@ -1171,41 +1171,59 @@ void play_controller::play_turn()
...
-		if(gui_->video().non_interactive()) {
-			LOG_AIT << " Player " << current_side() << ": " <<
-				current_team().villages().size() << " Villages" <<
-				std::endl;
-			ai_testing::log_turn_end(current_side());
+			ai_testing::log_turn_start(current_side());
+			play_side();
+			//ignore any changes to next_player_number_ that happen after the [end_turn] is sended to the server, otherwise we will get OOS.
+			next_player_number = gamestate_->next_player_number_;
+			assert(next_player_number <= 2 * int(gamestate().board_.teams().size()));
```
[the remark this answers is not in the document]
Comment (gfgtdf) [reply src/play_controller.cpp]: no this is a simple getter that does not have any side effects.

**Comment by GregoryLundberg** [comment]
No idea why it shows me deleting messages. I guess it means that Codacy updated/replaced its comment and since I set up Codacy on my personal fork, I'm getting blamed for it. I'll raise the issue with Codacy.

File: src/play_controller.cpp [hunk 2]
```diff
@@ -1170,41 +1170,59 @@ void play_controller::play_turn()
...
```
[the remark this answers is not in the document]
Comment (CelticMinstrel) [reply src/play_controller.cpp]: Pretty sure this function doesn't have side-effects that the code is depending on. Plus, Wesnoth builds with asserts enabled even in release mode, for some reason.

[the remark this answers is not in the document]
Comment (gfgtdf) [reply src/play_controller.cpp]: don't argue with the bot... , it will delete your and its post, and then recreate its allegation

**Pull request merged**